server/game/cards/agendas/thehousewiththereddoor.js

```diff
@@ -0,0 +1,50 @@
+const AgendaCard = require('../../agendacard.js');
+
+class TheHouseWithTheRedDoor extends AgendaCard {
+    constructor(owner, cardData) {
+        super(owner, cardData);
+
+        this.registerEvents(['onDecksPrepared']);
+    }
+
+    onDecksPrepared() {
+        this.game.promptForDeckSearch(this.controller, {
+            activePromptTitle: 'Select a location',
+            cardCondition: card => card.getType() === 'location' && !card.isLimited() && card.isUnique() && card.getPrintedCost() <= 3,
+            onSelect: (player, card) => this.cardSelected(player, card),
+            onCancel: player => this.doneSelecting(player),
+            source: this
+        });
+    }
+
+    cardSelected(player, card) {
+        this.startLocation = card;
+        player.putIntoPlay(card);
+
+        //Manually set card faceup to override the setup phase default facedown setting
+        card.facedown = false;
+
+        this.controller.cardsInPlayBeforeSetup.push(card);
+        this.game.addMessage('{0} uses {1} to search their deck and put {2} into play', player, this, card);
+    }
+
+    doneSelecting(player) {
+        this.game.addMessage('{0} uses {1} to search their deck, but does not put any card into play', player, this);
+    }
+
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            targetType: 'player',
+            effect: ability.effects.setSetupGold(4)
+        });
+
+        this.persistentEffect({
+            match: this.startLocation,
```

Comment:
I don't think this actually works. At the time `setupCardAbilities` is called (when the game creates the agenda's Card object but before the deck has been fully prepared), `this.startLocation` is undefined. It's later defined after the search prompt but that doesn't update the reference here. So at the time the card is created, it's fixed to match an undefined card (I don't know what side effects that would have).

I think you'll either need to do `match: card => card === this.startLocation`, which *will* update the reference as new cards come into play, or move the call to `persistentEffect` inside the `cardSelected` method (which seems a bit weird but should work as well)